itest.py

```diff
@@ -0,0 +1,832 @@
+                if plugin_choice.startswith(prefix):
+                    plugin_name = plugin_choice[len(prefix)+1:]
+
+            # Append plugin dir
+            named_args.append(('--plugin-package-directory', str(IRODS_PLUGIN_PACKAGE_DIR).format(os_name=os_choice, short_plugin_name=plugin_name)))
+        elif desired_test.stem == 'run_core_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
+        elif desired_test.stem == 'run_topology_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
+
+            topo_options = ['provider', 'consumer']
+            for option in topo_options:
+                print(option)
+
+            choice = input('Where ya wanna run the tests? ')
+            validate_choice(choice, topo_options)
+
+            bonus_args.append(choice)
+
+    else:
+        test_idx = len(test_args) 
+        try:
+            test_idx = test_args.index('--tests')
+        except ValueError:
+            ...
+        bonus_args = test_args[:test_idx]
+
+    arg_list = [sys.executable, str(desired_test),]
+    for arg, param in named_args:
+        arg_list.append(arg)
+
+        if isinstance(param, PurePath):
+            param = str(param).format(os_name=os_choice)
+
+        arg_list.append(param)
+
+    arg_list.extend(bonus_args)
+
+    failed_tests_list = None
+    if test_args is None:
+        tests_to_run = input('Please input test you wish to run (default: all): ').split()
+
+        if len(tests_to_run) != 0:
+            failed_tests_list = tests_to_run
+            failed_tests_list.insert(0, '--tests')
+    else:
+        failed_tests_list = test_args[test_idx:]
+
+    res = None
+    failed_tests_pattern = re.compile('(?<=List of failed tests:\n\t).*')
+
+    max_duplicate_attempts = 3
+    current_duplicate_attempt = 1
+    attempt = 0
+
+    while current_duplicate_attempt <= max_duplicate_attempts:
+        attempt += 1
+
+        logger.info(f'Running attempt=[{attempt!r}] with current_duplicate_attempt=[{current_duplicate_attempt!r} of max_duplicate_attempts=[{max_duplicate_attempts!r}]]')
+
+        if failed_tests_list:
+            running_arg_list = arg_list + failed_tests_list
+        else:
+            running_arg_list = arg_list
+
+        logger.debug(f"Executing running_arg_list=[{running_arg_list!r}]")
+
+        try:
+            #with subprocess.Popen(running_arg_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True) as res:
+            #    for line in res.stdout:
+            #        print(line, end='')
+
+            #res = stream_subprocess_output(running_arg_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
+            res = subprocess.run(running_arg_list, capture_output=True, text=True)
```

Comment:
Delete the 'streaming output' code for now, unless it can be implemented immediately.

Reply:
Removed.